components/Chat.tsx

```diff
@@ -63,34 +71,33 @@ export default function Chat() {
         throw new Error(`HTTP error! status: ${res.status}`);
       }
 
-      const { id } = await res.json()
+      const { id } = await res.json();
 
       const interval = window.setInterval(async () => {
         const pollRes = await fetch(`/api/chat/poll?job_id=${id}`, {
-          method: 'GET',
-        })
+          method: "GET",
+        });
         if (!res.ok) {
           setLoading(false);
-          console.error("Stopped polling")
-          window.clearInterval(interval)
-          return
+          console.error("Stopped polling");
+          window.clearInterval(interval);
+          return;
```

Comment:
move this to `finally` clause

Reply:
We can't use finally block to de-duplicate the cleanup code because when there's no error, we need the interval to continue.

Reply:
My bad, youre right